### ISSUE TITLE
[SP9] Won't wake up from suspend, boot up and restarts are fine.

### ISSUE THREAD

**Issue by tinywretch** [linux-surface/linux-surface#1224]
I'm new to linux so I apologise if this is a useless question but I really don't want to go back to windows. Whenever I close the type cover I can no longer wake the machine back up. It either takes a really long time or it just doesn't turn back on. It takes so long I beat a dark souls boss while waiting for it to wake up. Everything else works fine. I genuinely have no idea what is wrong but then again I'm new to all this. I would really appreciate if someone could help me because I start classes soon and I want a usable machine.

### Environment

- Hardware model: Surface Pro 9 2038 i7, 16gb RAM, 256gb SSD.
- Kernel version: 6.4.7-1.surface.fc38.x86_64
- Distribution: Fedora 38 Workstation

**Comment by tinywretch** [comment]
Alternatively so I can avoid any complications for the meantime, is there a way to prevent the type cover closing to do anything? In other words, if the type cover closes nothing happens and the screen stays on so I have to manually suspend by pressing the power button. That would be a satisfying solution to me.

**Comment by soyoh** [comment]
I have the same issue, but with ubuntu. It wont wake up at all, and sleep is just draining the battery (30% in 5h) 

To avoid the close lid [check this](https://github.com/linux-surface/linux-surface/wiki/Known-Issues-and-FAQ#suspend-aka-sleep-vs-lid-closingopening-events) or use gnome-tweaks and disable the "suspend when laptop lid is closed"

**Comment by derkrasseleo** [comment]
I am not sure if it was the new kernel (6.8.1) or maybe GNOME 46, but my SP9 seems to wake up normally again.

**Comment by derkrasseleo** [comment]
Aaaand I take it back, I let my SP9 sit for a while, and it didn't wake up from suspend, couldn't even turn it on properly. I had to hold the power button for a while and then press it again to start.

**Comment by inhumantsar** [comment]
My SP9 is also unable to wake up after suspend on NixOS with 6.9.12 and KDE Plasma 6, though I don't use the typecover so suspend only happens because of the power settings idle timer or manually using the OS menu to do it.

**Comment by xieve** [comment]
I'm experiencing this issue intermittently, and have not been able to discern a cause for it. I have applied [this workaround](https://github.com/linux-surface/linux-surface/wiki/Surface-Pro-9#acpi-interrupt-storm) to avoid the lid closing ACPI storms, which seems to have worked, as my kernel log says
```
kernel: PM: suspend entry (s2idle)
```
as the last line when suspend fails.

**Comment by inhumantsar** [comment]
this is fixed for me using kernel v6.11.4. 

in fact, it works more reliably than sleep did in windows

**Comment by derkrasseleo** [comment]
Anyone else experience this again after updating to surface-6.12.4?

**Comment by dx6nxr** [comment]
I'm not getting the 6.12.4 kernel and stuck with 6.12.3, but the sp9 still fails to wake up.

**Comment by xieve** [comment]
> Anyone else experience this again after updating to surface-6.12.4?

Yes, I'm on `surface-6.12.8` currently via nixos-hardware. When suspend works, it always works until the system is shutdown completely, i.e. for a given "session", it either works reproducibly or it doesn't work. Notably, when it doesn't work, none of the hardware buttons work, which makes me think that the actual issue isn't waking from sleep, but that the hardware buttons don't work on some boots (roughly 60-70% I'd say). I have not yet tested this hypothesis though.
